service/history/timerQueueProcessorBase.go

```diff
@@ -711,31 +706,10 @@ func (t *timerQueueProcessorBase) deleteWorkflowHistory(task *persistence.TimerT
 }
 
 func (t *timerQueueProcessorBase) deleteWorkflowVisibility(task *persistence.TimerTaskInfo) error {
-	switch {
-	case t.visibilityProducer != nil:
-		msg := getVisibilityMessageForDeletion(task.DomainID, task.WorkflowID, task.RunID, task.GetTaskID())
-		op := func() error {
-			return t.visibilityProducer.Publish(msg)
-		}
-		return backoff.Retry(op, kafkaOperationRetryPolicy, common.IsKafkaTransientError)
-	default:
-		op := func() error {
-			return t.historyService.DeleteExecutionFromVisibility(task.DomainID, task.RunID)
-		}
-		return backoff.Retry(op, persistenceOperationRetryPolicy, common.IsPersistenceTransientError)
-	}
-}
-
-func getVisibilityMessageForDeletion(domainID, workflowID, runID string, docVersion int64) *indexer.Message {
-	msgType := indexer.MessageTypeDelete
-	msg := &indexer.Message{
-		MessageType: &msgType,
-		DomainID:    common.StringPtr(domainID),
-		WorkflowID:  common.StringPtr(workflowID),
-		RunID:       common.StringPtr(runID),
-		Version:     common.Int64Ptr(docVersion),
+	op := func() error {
+		return t.historyService.DeleteExecutionFromVisibility(task)
```

Comment:
If deletion from es is successful but deletion from db failed, will the deletion from es fail in the next attempt?

Reply:
No it won't. It just publish a duplicate deletion event to kafka, and indexer to ES will handle such duplicates.